### DISCUSSION TITLE
Dependencies Resolution

### DISCUSSION THREAD

**Discussion by naveeng2402** [lautr/initium-nuxt#409] [Ideas]
Hi @lautr really interested by your work and excited, but I'm having several doubts on how the dependencies are managed between the vue app and nuxt app,

You've using tailwind in both of them but in nuxt we have tailwindcss module is it possible to use that and how can we synchronise the tailwindconfig.

Can you shed some light on how dependencies between these projects are integrated.

**Comment by lautr** [comment]
Hi @naveeng2402,

Thank you for expressing interest in my project. I originally chose to use Tailwind directly, rather than utilizing a module, in order to simplify the process of ensuring that both the design system and applications utilize the same Tailwind version.

However, I am currently working on adding Nuxt Layer support to facilitate the integration of multiple Nuxt applications. In doing so, I plan to transition to Nuxt Tailwind, as it streamlines various tasks, and i will also centralize the Tailwind configuration. Although this enhancement is still a work in progress, you can take a look: https://github.com/lautr/initium-nuxt/pull/349.

Regarding dependencies, I designed the approach so that both the applications and the design system can function independently, without any indirect or hidden dependencies. The sole exception is the inclusion of ESLint and TypeScript, which are defined at the monorepo level. I made this decision as a compromise in order to improve the overall developer experience.

If you have any additional questions or suggestions for enhancing the starter kit, please don't hesitate.

**Comment by naveeng2402** [comment]
Hi @lautr, exploring on the setup and tried a similar approach on an actual product and now I'm here with some suggestions

- Get a way to integrate iconify for nuxt we have `Nuxt Icon` and for vue, we have `iconify/vue` but also found that iconify also has a web component. I'm not too familiar with them so I'm unable to make the intelliscense work.
- We may need to figure out how to add `nuxt-image` support which not only has amazing optimisation features but also integrates seamlessly with cloud images from various sources such as cloudinary
- Additonally we can also add a composables directory to the vue app, so that we can able to create renderless components and have a robust design system
- Similar to `vueuse` we can opt in to include `vuemacros`

**Reply by lautr** [reply]
Hello @naveeng2402,

I appreciate your insights, and I'm grateful for the suggestions you've shared. I agree that incorporating macros and a composable directory within the design system would be a valuable enhancement, so I'll move forward with those implementations.

Regarding the addition of more Nuxt Modules, I'm somewhat hesitant at the moment. While I understand the potential benefits, I believe that the Image & Icon integrations are relatively straightforward, and including them directly in the template might not offer substantial advantages. Additionally, it could make the solution more opinionated than necessary.

After completing the Multi App / Nuxt Layer integration, I plan to revisit the idea of incorporating additional modules. At a minimum, it could be worthwhile to provide a concise guide on integrating the most common Nuxt Modules, which could offer support and value to users.

Once again, thank you for your input, and please feel free to share any further thoughts or suggestions.

**Reply by naveeng2402** [reply]
Hi @lautr,

I'm excited to hear about the progress on the nuxt later integration. And I'm really waiting to use the template in a large scale cloud app soon.
